segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/stat/AutoScaleProcessor.java

```diff
@@ -192,113 +202,99 @@ static boolean hasTlsEnabled(@NonNull final URI controllerURI) {
+        Pair<Long, Long> pair = cache.getIfPresent(streamSegmentName);
+        long lastRequestTs = 0;
 
-            if (pair != null && pair.getValue() != null) {
-                lastRequestTs = pair.getValue();
-            }
+        if (pair != null && pair.getValue() != null) {
+            lastRequestTs = pair.getValue();
+        }
 
-            long timestamp = System.currentTimeMillis();
-            long requestId = requestIdGenerator.get();
-            if (timestamp - lastRequestTs > configuration.getMuteDuration().toMillis()) {
-                log.info(requestId, "sending request for scale down for {}", streamSegmentName);
-
-                Segment segment = Segment.fromScopedName(streamSegmentName);
-                AutoScaleEvent event = new AutoScaleEvent(segment.getScope(), segment.getStreamName(), segment.getSegmentId(),
-                        AutoScaleEvent.DOWN, timestamp, 0, silent, requestId);
-                writeRequest(event, () -> {
-                    if (!silent) {
-                        // mute only scale downs
-                        cache.put(streamSegmentName, new ImmutablePair<>(0L, timestamp));
-                    }
-                });
-            }
+        long timestamp = getTimeMillis();
+        long requestId = requestIdGenerator.get();
+        if (timestamp - lastRequestTs > configuration.getMuteDuration().toMillis()) {
+            log.info(requestId, "sending request for scale down for {}", streamSegmentName);
+
+            Segment segment = Segment.fromScopedName(streamSegmentName);
+            AutoScaleEvent event = new AutoScaleEvent(segment.getScope(), segment.getStreamName(), segment.getSegmentId(),
+                    AutoScaleEvent.DOWN, timestamp, 0, silent, requestId);
+            writeRequest(event, () -> {
+                if (!silent) {
+                    // mute only scale downs
+                    cache.put(streamSegmentName, new ImmutablePair<>(0L, timestamp));
+                }
+            });
         }
     }
 
     private void writeRequest(AutoScaleEvent event, Runnable successCallback) {
-        val writer = this.writer.get();
-        if (writer == null) {
-            log.warn(event.getRequestId(), "Writer not bootstrapped; unable to post Scale Event {}.", event);
-        } else {
-            writer.writeEvent(event.getKey(), event)
+        getWriter().thenCompose(w -> w.writeEvent(event.getKey(), event)
                     .whenComplete((r, e) -> {
                         if (e != null) {
                             log.error(event.getRequestId(), "Unable to post Scale Event to RequestStream '{}'.",
                                     this.configuration.getInternalRequestStream(), e);
                         } else {
-                            log.debug(event.getRequestId(), "Scale Event posted successfully: {}.", event);
+                            log.info(event.getRequestId(), "Scale Event posted successfully: {}.", event);
```

Comment:
Please revert this. This will result in too much garbage in the segment store logs.